libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/ThemeSwitcher.java

```diff
@@ -56,11 +67,10 @@ static void toggleTheme(Activity activity) {
    * @param map     the style will be set on
    */
   static void setMapStyle(Context context, MapboxMap map, MapboxMap.OnStyleLoadedListener listener) {
-    SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(context);
-    boolean darkThemeEnabled = preferences.getBoolean(context.getString(R.string.dark_theme_enabled), false);
-    String nightThemeUrl = context.getString(R.string.navigation_guidance_night_v2);
-    String dayThemeUrl = context.getString(R.string.navigation_guidance_day_v2);
-    map.setStyleUrl(darkThemeEnabled ? nightThemeUrl : dayThemeUrl, listener);
+    TypedValue mapStyleAttr = new TypedValue();
```

Comment:
Could we extract this block of code into a shared method in order to avoid duplication?

:eyes: methods `retrieveNavigationViewPrimaryColor`, `retrieveNavigationViewSecondaryColor`, `retrieveNavigationViewBannerBackgroundColor`, `retrieveNavigationViewBannerManeuverPrimaryColor`, `retrieveNavigationViewBannerManeuverSecondaryColor`, `retrieveNavigationViewProgressColor` and `retrieveNavigationViewProgressBackgroundColor`